### PR TITLE
Deviceid cleanup

### PR DESCRIPTION
**Pull request by ryandoherty**
I started with stripping whitespace, but it snowballed into adding an error field to the settings view. Now (most) settings have a space to show error messages when the user changes settings.

Added better device id validation. Checks to see if the value entered is alphanumeric.

### CODE REVIEW

File: autosportlabs/racecapture/views/configuration/rcp/telemetryconfigview.py [hunk 1]
```diff
@@ -29,10 +32,17 @@ def __init__(self, **kwargs):
...
+            value = strip_whitespace(value)
+            self.ids.device_id.ids.error.text = ''
+            if len(value) > 0:
+                if self.validate_device_id(value):
+                    kvFind(self, 'rcid', 'deviceId').setValue(value)
+                    self.connectivityConfig.telemetryConfig.deviceId = value
+                    self.connectivityConfig.stale = True
+                    self.dispatch('on_modified')
+                else:
+                    self.ids.device_id.ids.error.text = 'Invalid device id. Id must only be numbers and letters.'
```
Comment (brentpicasso) [inline autosportlabs/racecapture/views/configuration/rcp/telemetryconfigview.py]: if you haven't, please test this on phone form factor screen (there's a launcher for that)

File: settingsview.kv [hunk 2]
```diff
@@ -1,31 +1,43 @@
...
+    BoxLayout:
+        rcid: 'fieldContainer'
+        orientation: 'horizontal'
+        BoxLayout:
+            orientation: 'vertical'
+            spacing: 5
+            size_hint: (0.6, 1.0)
+            FieldLabel:
+                size_hint: (1, 1.0)
+                rcid: 'fieldLabel'
```
Comment (brentpicasso) [inline settingsview.kv]: rcid shouldn't be used going forward - we should use kivy's id mechanism whenever we update or add new code

File: autosportlabs/racecapture/views/configuration/rcp/telemetryconfigview.py [hunk 3]
```diff
@@ -44,8 +54,6 @@ def on_config_updated(self, rcpCfg):
         kvFind(self, 'rcid', 'bgStream').setValue(connectivityConfig.telemetryConfig.backgroundStreaming)
```
Comment (brentpicasso) [inline autosportlabs/racecapture/views/configuration/rcp/telemetryconfigview.py]: you'd use self.ids.bgStream ... instead

Comment (brentpicasso) [inline autosportlabs/racecapture/views/configuration/rcp/telemetryconfigview.py]: just realized this is legacy code.

File: settingsview.kv [hunk 4]
```diff
@@ -1,31 +1,43 @@
...
+                text: ''
+        BoxLayout:
+            padding: [0, 10, 0, 10]
+            size_hint: (0.4, 1.0)
+            orientation: 'vertical'
+            halign: 'left'
+            spacing: 5
+            BoxLayout:
+                padding: [10,0]
+                rcid: 'control'
```
Comment (brentpicasso) [inline settingsview.kv]: does this control rcid get used?

File: settingsview.kv (hunk 2 above)
Comment (ryandoherty) [inline settingsview.kv]: The rcid isn't new, just moved some old code around.

File: settingsview.kv (hunk 4 above)
Comment (ryandoherty) [inline settingsview.kv]: I think so, it's old code, didn't dig around.

File: autosportlabs/racecapture/views/configuration/rcp/telemetryconfigview.py (hunk 1 above)
Comment (ryandoherty) [inline autosportlabs/racecapture/views/configuration/rcp/telemetryconfigview.py]: :+1:

**Comment by brentpicasso** [comment]
horizontal proportions are off for settingsview is off - the left side label used to have the majority of the space. Investigating..

**Comment by brentpicasso** [comment]
suggest proportion change for settingsview.kv 

index 9442c47..e387b3d 100644
--- a/settingsview.kv
+++ b/settingsview.kv
@@ -7,7 +7,7 @@
         BoxLayout:
             orientation: 'vertical'
             spacing: 5
-            size_hint: (0.6, 1.0)
+            size_hint: (0.65, 1.0)
             FieldLabel:
                 size_hint: (1, 1.0)
                 rcid: 'fieldLabel'
@@ -21,7 +21,7 @@
                 text: ''
         BoxLayout:
             padding: [0, 10, 0, 10]
-            size_hint: (0.4, 1.0)
+            size_hint: (0.35, 1.0)
             orientation: 'vertical'
             halign: 'left'
             spacing: 5

File: autosportlabs/racecapture/views/configuration/rcp/telemetryconfigview.py (hunk 1 above)
Comment (brentpicasso) [inline autosportlabs/racecapture/views/configuration/rcp/telemetryconfigview.py]: suggest:
error text should be red
make concise: Id may contain numbers and letters

Comment (brentpicasso) [inline autosportlabs/racecapture/views/configuration/rcp/telemetryconfigview.py]: errr... 

"Id may only contain numbers and letters"

**Comment by brentpicasso** [comment]
other than the suggested proportions and error text changes, :shipit:

**Pull request merged**